cloudinit/config/cc_refresh_rmc_and_interface.py

```diff
@@ -0,0 +1,134 @@
+  is used for communication between RMC daemon and PowerVM hypervisor.
+
+**Internal name:** per always
+
+**Supported distros:** RHEL
+
+"""
+
+from cloudinit import log as logging
+from cloudinit.settings import PER_ALWAYS
+from cloudinit import util
+from cloudinit import subp
+from cloudinit import netinfo
+
+import errno
+
+frequency = PER_ALWAYS
+
+LOG = logging.getLogger(__name__)
+# Ensure that /opt/rsct/bin has been added to standard PATH of the
+# distro. The symlink to rmcctrl is /usr/sbin/rsct/bin/rmcctrl .
+RMCCTRL = 'rmcctrl'
+SED = 'sed'
+
+
+def handle(name, _cfg, _cloud, _log, _args):
+    info = netinfo.netdev_info()
+    # Setting NM_CONTROLLED=no for IPv6 interface
+    # making it down and up
+
+    LOG.debug('Making the IPv6 up explicitly. '
+              'Ensuring IPv6 interface is not being handled by NetworkManager'
+              ' and it is  restarted to re-establish the communication with '
+              'the hypervisor')
+    for iface, data in info.items():
+        if iface == "lo":
+            LOG.debug('Skipping localhost interface')
+            continue
+        if len(data.get("ipv4", [])) != 0:
+            # skip this interface, as it has ipv4 addrs
+            continue
+        try:
+            disable_NM_ipv6(iface)
+            refresh_ipv6(iface)
+        except BaseException:
+            util.logexc(LOG, 'Interface is not up')
+
+    refresh_rmc()
+
+
+def refresh_ipv6(interface):
+    # IPv6 interface is explicitly brought up, subsequent to which the
+    # RMC services are restarted to re-establish the communication with
+    # the hypervisor.
+    subp.subp(['ip', 'link', 'set', interface, 'down'])
+    subp.subp(['ip', 'link', 'set', interface, 'up'])
+
+
+def disable_NM_ipv6(interface):
+    # Ensuring that the communication b/w the hypervisor and VM is not
+    # interrupted due to NetworkManager. For this purpose, as part of
+    # this function, the NM_CONTROLLED is explicitly set to No for IPV6
+    # interface and NetworkManager is restarted.
+    interface_file = '/etc/sysconfig/network-scripts/ifcfg-' + interface
+    ipv6_details = ['/^IPV6ADDR/d', '/^IPADDR6/d', '/^IPV6INIT/d']
+    try:
+        contents = util.load_file(interface_file)
+    except IOError as e:
+        if e.errno == errno.ENOENT:
+            LOG.debug("IPv6 interface file %s does not exist\n",
+                      interface_file)
+        else:
+            raise e
+
+    if 'IPV6INIT' not in contents:
+        LOG.debug("Interface file %s did not have IPV6INIT", interface_file)
+        return
+
+    LOG.debug("Editing interface file %s for iface %s",
+              interface_file, interface)
+
+    if contents.find('NM_CONTROLLED') < 0:
+        with open(interface_file, "a") as file_object:
+            file_object.write("NM_CONTROLLED=no")
+    else:
+        subp.subp([SED, '-i', 's/NM_CONTROLLED=yes/NM_CONTROLLED=no/g',
+                   interface_file])
+
```

Comment:
you've got 'contents' now, so this isn't much more work to do this without 'sed'.

```python
func adjust_ipv6_iface_file(path):

    contents = util.load_file(path)
    # we drop any NM_CONTROLLED or IPV6 lines.
    dropline = lambda l: (
        l.startswith("IPV6ADDR") or
        l.startswith("IPADDR6") or
        l.startswith("IPV6INIT") or
        l.startswith("NM_CONTROLLED="))

    lines = contents.splitlines()
    lines = [l for l in lines if not dropline(l)]
    lines.append("NM_CONTROLLED=no")

    with open(interface_file, "w") as fp:
        fp.write("\n".join(lines) + "\n")
```

That is then really easy to unit test also, by just writing a temp file,
then calling the function, then reading it back and testing contents.

You can make testing even easier by passing in contents and returning
contents (doign the reading and writing out side of the function)